query_engine/src/datafusion_impl/task_context.rs

```diff
@@ -225,7 +226,7 @@ impl RemotePhysicalPlanExecutor for RemotePhysicalPlanExecutorImpl {
 
             // Remote execute.
             let stream = remote_engine
-                .execute_physical_plan(request)
+                .execute_physical_plan(request, metrics_collector)
```

Comment:
Remote engine is just used for sending something, something like this should be placed in the interface.